### PR TITLE
Add k8s deployment config

### PR DESCRIPTION
**Pull request by RammusXu**

### CODE REVIEW

**Review by aweris — COMMENTED** [review]
There are some issues when you want to use runners with Kubernetes.

- Github action runners have a self-update feature. It means when they update servers your pods will be stuck in the update cycle.  Please check the links below for the problem and possible solutions.
https://github.community/t5/GitHub-Actions/self-hosted-runner-update-protocol/td-p/37930
 https://github.com/gdraheim/docker-systemctl-replacement
- You need to persist runner config somehow and reuse it. Since Tokens are time-based they'll expire quickly and when that happens your runner will get an error for the expired token when Kubernetes recreate existing POD
 I think saving `.runner`, `.credentials`, `.credentials_rsaparams` is enough.
- Adding `--replace` option to `/config.sh`  is a good idea for avoiding `Runner with the same name exist` error in the configuration when you want to configure new runner with the same name

File: deploy/kustomize/statefulset-action-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      app: github-runner-action-demo
+  template:
+    metadata:
+      labels:
+        app: github-runner-action-demo
+    spec:
+      containers:
+      - name: agent
+        image: swaglive/github-runner:2.163.1-ubuntu18.04
+        imagePullPolicy: Always
```
Comment (allanlei) [inline deploy/kustomize/statefulset-action-demo.yaml]: Probably don't need this since its tagged with version

Comment (RammusXu) [reply deploy/kustomize/statefulset-action-demo.yaml]: Yes. It's for debugging. I'll remove it.

**Comment by RammusXu** [comment]
> * Github action runners have a self-update feature. It means when they update servers your pods will be stuck in the update cycle.  Please check the links below for the problem and possible solutions.
>   https://github.community/t5/GitHub-Actions/self-hosted-runner-update-protocol/td-p/37930
>   https://github.com/gdraheim/docker-systemctl-replacement

I prefer to let kubernetes handle the stuck and recreate a pod.

> * You need to persist runner config somehow and reuse it. Since Tokens are time-based they'll expire quickly and when that happens your runner will get an error for the expired token when Kubernetes recreate existing POD
>   I think saving `.runner`, `.credentials`, `.credentials_rsaparams` is enough.

I updated `entrypoint.sh` and `statefulset.yaml`
- https://github.com/swaglive/docker-github-actions-runner/blob/master/entrypoint.sh#L19-L21
- https://github.com/swaglive/docker-github-actions-runner/pull/4/commits/8fb6a3397b2b12da0c27d706bd6fef5e54e966d5#diff-8117c090fb3deb9de126ef39b4c8b350R51-R52

> * Adding `--replace` option to `/config.sh`  is a good idea for avoiding `Runner with the same name exist` error in the configuration when you want to configure new runner with the same name

ok. It's nice to have.
#6

**Comment by aweris** [comment]
> I prefer to let kubernetes handle the stuck and recreate a pod.

Agree with you about this but the problem is when github-actions-runner self-update and try to restart process kubernetes will interrupt and start new pod with the old version. So your runners because out of order till you update latest version. 

For testing this scenario you can downgrade your runner version in the docker image

**Pull request merged**